### ISSUE TITLE
change torrent location does not trigger 'recently-active'

### ISSUE THREAD

**Issue by reardonia** [transmission/transmission#5996]
### What is the issue?

Logging this quickly, will try to followup with more details.

In Tr3, any change to torrent location (via RPC either "torrent-set" or "torrent-set-location") would trigger an update such that "torrent-get(recently-active)" would reveal the new location.  In Tr4 this is no longer true; only an explicit torrent-get(ids) will reveal the updated location.

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.4

**Comment by tearfur** [comment]
Does it? It doesn't seem like it judging from the code alone.

**Comment by reardonia** [comment]
Yes, I am positive that Tr3 does the right thing here and Tr4 does not.  Try it.  On a paused torrent, try torrent-set-location (can just use transmission-remote for this) while monitoring torrent-get(recently-active).

**Comment by tearfur** [comment]
I must be missing something obvious here. Isn't `recently-active` only affected by `anyDate`? `tr_torrentMarkEdited()` modifies `editDate` and `tr_torrentSetDirty()` modifies `isDirty`.

https://github.com/transmission/transmission/blob/bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd/libtransmission/rpcimpl.c#L159-L174

**Comment by reardonia** [comment]
I don't know; unlikely I can chase further into the code.  For now, just take my issue report as an end-user issue: Tr3 works, Tr4 does not. It is straightforward to repro.

**Comment by tearfur** [comment]
Revisited this and finally found where the difference was. I'm not sure if the behaviour in `3.00` was intentional though, I think only @ckerr has the answer to this.

### 3.00

`tr_torrent.anyDate` gets updated like this:

1. `setLocation()` calls `tr_verifyRemove()`.

   https://github.com/transmission/transmission/blob/bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd/libtransmission/torrent.c#L3272

2. `tr_verifyRemove()` calls `tr_torrentSetVerifyState()` to set the torrent's verify state to "none", as long as the torrent isn't currently verifying, *even if the torrent was not in the verify queue*.

   https://github.com/transmission/transmission/blob/bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd/libtransmission/verify.c#L317-L321

3. `tr_torrent.anyDate` gets updated.

   https://github.com/transmission/transmission/blob/bb6b5a062ee594dfd4b7a12a6b6e860c43849bfd/libtransmission/torrent.c#L1262-L1269

### 4.x.x

Setting torrent location also removes the torrent from the verify queue like `3.00` does:

https://github.com/transmission/transmission/blob/1ec6b2b2328c114075171917ef417d13c35cc495/libtransmission/torrent.cc#L1106

The key difference is here, *the torrent verify state does not get updated if the torrent was not in the verify queue*:

https://github.com/transmission/transmission/blob/1ec6b2b2328c114075171917ef417d13c35cc495/libtransmission/verify.cc#L181-L190

`tr_torrent::VerifyMediator::on_verify_done()` --> `tr_torrent::set_verify_state()` --> `tr_torrent::mark_changed()`

https://github.com/transmission/transmission/blob/1ec6b2b2328c114075171917ef417d13c35cc495/libtransmission/torrent.cc#L1569-L1576
